### PR TITLE
C++: Support more builtin operations

### PR DESCRIPTION
**Pull request by jketema**

### CODE REVIEW

File: cpp/ql/lib/semmle/code/cpp/exprs/BuiltInOperations.qll [hunk 1]
```diff
@@ -1554,14 +1754,135 @@ class BuiltInBitCast extends BuiltInOperation, @builtinbitcast {
...
+ * Returns `true` if a type is a trivial type.
+ * ```
+ * template<typename _Tp>
+ *   struct reference_converts_from_temporary
+ *   : public integral_constant<bool, __reference_converts_from_temporary(_Tp)>
+ *   {};
+ * ```
+ */
+class BuiltInOperationReferenceCovertsFromTemporary extends BuiltInOperation,
+  @referenceconstructsfromtemporary
```
Comment (MathiasVP) [inline cpp/ql/lib/semmle/code/cpp/exprs/BuiltInOperations.qll:1803]: This should be `@referenceconvertsfromtemporary` right?

**Review by MathiasVP — COMMENTED** [review]
One comment, but otherwise this LGTM!

File: cpp/ql/lib/semmle/code/cpp/exprs/BuiltInOperations.qll (hunk 1 above)
Comment (jketema) [reply cpp/ql/lib/semmle/code/cpp/exprs/BuiltInOperations.qll]: Fixed in ed42c3cd6f508e988cbae3e1f0e1139be20087bd

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Review by MathiasVP — APPROVED** [review]

**Pull request merged**